.github/workflows/icu_ci.yml

```diff
@@ -498,18 +498,97 @@ jobs:
+          if [ "12" != `egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h| wc -l` ]; then
+              echo "More than %s UCONFIG_NO_* defined in uconfig.h, please adjust uconfig_cppflags above to include newly added flag";
+              echo "Currently UCONFIG_NO_* defined in uconfig.h:";
+              egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h;
+              exit -1
+          fi
+      - name: Build and Test
+        env:
+          CPPFLAGS: ${{ matrix.uconfig_cppflags }}
+        run: |
+          cd icu4c/source/;
+          ./runConfigureICU Linux;
+          make -j2 tests;
 
   # Run header tests with UCONFIG_NO_XXX variations.
-  uconfig_variation-check-all-header-tests:
+  uconfig-header-tests:
     runs-on: ubuntu-latest
+    strategy:
+      # "fail-fast: false" let other jobs keep running even if the test break in some other uconfig.
+      fail-fast: false
+      matrix:
+        uconfig_cppflags:
+         # Ignore the following two.
+         # - "-DUCONFIG_NO_FILE_IO=1"
+         # - "-DUCONFIG_NO_CONVERSION=1"
+         - "-DUCONFIG_NO_LEGACY_CONVERSION=1"
+         - "-DUCONFIG_NO_NORMALIZATION=1"
+         - "-DUCONFIG_NO_BREAK_ITERATION=1"
+         - "-DUCONFIG_NO_IDNA=1"
+         - "-DUCONFIG_NO_COLLATION=1"
+         - "-DUCONFIG_NO_FORMATTING=1"
+         - "-DUCONFIG_NO_TRANSLITERATION=1"
+         - "-DUCONFIG_NO_REGULAR_EXPRESSIONS=1"
+         - "-DUCONFIG_NO_SERVICN=1"
+         - "-DUCONFIG_NO_FILTERED_BREAK_ITERATION=1"
+         # Turn on all the define in one test
+         - "-DUCONFIG_NO_LEGACY_CONVERSION=1 -DUCONFIG_NO_NORMALIZATION=1 -DUCONFIG_NO_BREAK_ITERATION=1 -DUCONFIG_NO_IDNA=1 -DUCONFIG_NO_COLLATION=1 -DUCONFIG_NO_FORMATTING=1 -DUCONFIG_NO_TRANSLITERATION=1 -DUCONFIG_NO_REGULAR_EXPRESSIONS=1 -DUCONFIG_NO_SERVICN=1 -DUCONFIG_NO_FILTERED_BREAK_ITERATION=1"
     steps:
-      - uses: actions/checkout@v2
-      - run: python3 tools/scripts/uconfig_vars_test.py -p
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Verify no additional new UCONFIG_NO_xxx added
+        run: |
+          # Test we have exact 12 "#ifndef UCONFIG_NO_" in uconfig.h. If the number change, we need to also
+          # adjust the uconfig_cppflags above to include the new one and update the "12" below.
+          if [ "12" != `egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h| wc -l` ]; then
+              echo "More than %s UCONFIG_NO_* defined in uconfig.h, please adjust uconfig_cppflags above to include newly added flag";
+              echo "Currently UCONFIG_NO_* defined in uconfig.h:";
+              egrep "#ifndef UCONFIG_NO_" icu4c/source/common/unicode/uconfig.h;
+              exit -1
+          fi
+      - name: Build and Install
+        run: |
+          cd icu4c/source/;
+          mkdir /tmp/icu_cnfg;
+          ./runConfigureICU Linux --prefix=/tmp/icu_cnfg;
```

Comment:
Should we `make clean > /dev/null` between runs?

Reply:
there are no "between" runs. All runs are done in parallel in different instance. Nothing in between and no relationship nor dependency on each other.